x-pack/plugins/security_solution/public/common/components/inspect/use_inspect.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import { useDeepEqualSelector } from '../../hooks/use_selector';
+import { inputsSelectors } from '../../store';
+import { inputsActions } from '../../store/actions';
+import { InputsModelId } from '../../store/inputs/constants';
+
+interface UseInspectModalProps {
+  inputId?: InputsModelId;
+  inspectIndex?: number;
+  isDisabled?: boolean;
+  multiple?: boolean;
+  onClick?: () => void;
+  onCloseInspect?: () => void;
+  queryId: string;
+  vizType?: string | null;
+}
+
+export const useInspect = ({
```

Comment:
Nice to have this `useInspect` hook 👍 
The only thing is this `vizType` prop seems too specific to visualizations to be in a common inspect hook and in its internal state, since it will only be used by a small fraction of all inspect buttons. 
I understand the problem it solves, but maybe the issue can be also solved by having a local boolean state (`const [isModalOpen, setModalOpen] = useState(false);`) in the _VisualizationActions_, and bypass the open and close functions with:
  ```
  const onOpenInspectModal = useCallback(() => {
    handleInspectButtonClick();
    setModalOpen(true);
  }, [handleInspectButtonClick]);
  
  const onCloseInspectModal = useCallback(() => {
    handleCloseInspectModal();
    setModalOpen(false);
  }, [handleCloseInspectModal]);
```
This way every _VisualizationActions_ will have its own show/hide control without polluting the `useInspect` state. Does it makes sense?

Reply:
yes, thank you, that's a smart idea! Implemented in https://github.com/elastic/kibana/pull/126507/commits/13fdc19368530b2383336f591aae80fce5ff8760